src/structarray.jl

```diff
@@ -441,3 +441,6 @@ BroadcastStyle(::Type{SA}) where SA<:StructArray = StructArrayStyle{typeof(cst(S
 
 Base.similar(bc::Broadcasted{StructArrayStyle{S}}, ::Type{ElType}) where {S<:DefaultArrayStyle,N,ElType} =
     isstructtype(ElType) ? similar(StructArray{ElType}, axes(bc)) : similar(Array{ElType}, axes(bc))
+
+# for aliasing analysis during broadcast
+Base.dataids(u::StructArray) = map(Base.dataids, values(components(u)))
```

Comment:
```suggestion
Base.dataids(u::StructArray) = mapreduce((a, b) -> (a..., b...), Base.dataids, values(components(u)), init=())
```

From the docs of `Base.dataids`:

```
  Base.dataids(A::AbstractArray)

  Return a tuple of UInts that represent the mutable data segments of an array.

  Custom arrays that would like to opt-in to aliasing detection of their component parts can specialize this method to
  return the concatenation of the dataids of their component parts. A typical definition for an array that wraps a
  parent is Base.dataids(C::CustomArray) = dataids(C.parent).
```

So I imagine you should concatenate the result instead of returning a tuple of tuples?

Reply:
Right, good point. Thanks!